builtin/providers/aws/resource_aws_cloudfront_web_distribution_test.go

```diff
@@ -0,0 +1,118 @@
+package aws
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccCloudFront(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudFrontDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSCloudFrontConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudfrontInitial(
+						"aws_cloudfront_web_distribution.main",
+					),
+				),
+			},
+			resource.TestStep{
+				Config: testAccAWSCloudFrontUpdate,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudfrontSecondary(
```

Comment:
Can we possibly follow the pattern from other acceptance tests here to make things a bit more readable, please?
i.e. use one function for checking the existence and other function for checking number of aliases & enabled status?

`Initial` & `Secondary` doesn't quite explain what exactly is happening inside the test.